testsuites/syncgateway/functional/tests/test_log_redaction.py

```diff
@@ -0,0 +1,454 @@
+    sg_url = cluster_hosts["sync_gateways"][0]["public"]
+    sg_ip = host_for_url(sg_admin_url)
+    sg_db = "db"
+    num_of_docs = 10
+    if get_sync_gateway_version(sg_ip)[0] < "2.1":
+        pytest.skip("log redaction feature not available for version < 2.1 ")
+
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+
+    # Modifying log redaction level to partial
+    temp_cluster_config = copy_to_temp_conf(cluster_config, mode)
+    persist_cluster_config_environment_prop(temp_cluster_config, 'redactlevel', "partial", property_name_check=False)
+    cluster = Cluster(config=temp_cluster_config)
+    cluster.reset(sg_config_path=sg_conf)
+
+    # Get sync_gateway host
+    cluster = load_cluster_config_json(cluster_config)
+    sg_host = cluster["sync_gateways"][0]["ip"]
+
+    # 3. Create user in sync_gateway
+    sg_client = MobileRestClient()
+    channels = ["log-redaction"]
+    sg_client.create_user(url=sg_admin_url, db=sg_db, name='autotest', password='validkey', channels=channels)
+    autouser_session = sg_client.create_session(url=sg_admin_url, db=sg_db, name='autotest', password='validkey')
+
+    # 4. Create docs with xattrs
+    sgdoc_bodies = document.create_docs(doc_id_prefix='sg_docs', number=num_of_docs,
+                                        attachments_generator=attachment.generate_2_png_10_10, channels=channels)
+    sg_client.add_bulk_docs(url=sg_url, db=sg_db, docs=sgdoc_bodies, auth=autouser_session)
+    assert len(sgdoc_bodies) == num_of_docs
+
+    # should throw an error when trying with upload host without upload
+    upload_host = "https://s3.amazonaws.com/cb-customers"
+    resp = sg_client.sgCollect_restCall(sg_host, redact_level="partial", upload_host=upload_host)
+    assert "Invalid options used for sgcollect_info: upload must be set to true if upload_host is specified" in resp.content
+    customer = "customer-name"
+    ticket = "123"
+
+    # should throw an error when trying with customer without upload parameter
+    resp = sg_client.sgCollect_restCall(sg_host, redact_level="partial", customer=customer)
+    assert "Invalid options used for sgcollect_info: upload must be set to true if customer is specified" in resp.content
+
+    # should throw an error when trying ticket without upload
+    resp = sg_client.sgCollect_restCall(sg_host, redact_level="partial", ticket=ticket)
+    assert "Invalid options used for sgcollect_info: upload must be set to true if ticket is specified" in resp.content
+
+    # should throw an error when trying with upload_host, customer and ticket without upload
+    resp = sg_client.sgCollect_restCall(sg_host, redact_level="partial", upload_host=upload_host, customer=customer, ticket=ticket)
+    assert "Invalid options used for sgcollect_info: upload must be set to true if upload_host is specified" in resp.content
+
+    # should throw an error when trying with output dir which does not exist
+    output_dir = "/abc"
```

Comment:
is / intended?

Reply:
yes, need to test with directory which does not exist